pkg/nmi/server/server.go

```diff
@@ -373,7 +376,14 @@ func (s *Server) msiHandler(w http.ResponseWriter, r *http.Request) (ns string)
 	podID, err := s.TokenClient.GetIdentities(r.Context(), podns, podname, tokenRequest.ClientID, tokenRequest.ResourceID)
 	if err != nil {
 		klog.Errorf("failed to get matching identities for pod: %s/%s, error: %+v", podns, podname, err)
-		http.Error(w, err.Error(), http.StatusNotFound)
+		httpErrorCode := http.StatusFound
```

Comment:
Is this supposed to be `http.StatusNotFound`

Reply:
That is the default value today. We're keeping the default behavior as is. If the feature is enabled, then we'll set it to `http.StatusServiceUnavailable` and add the `Retry-After` header.

Reply:
Updated!